src/Swarrot/Broker/MessagePublisher/PeclPackageMessagePublisher.php

```diff
@@ -11,14 +11,59 @@ class PeclPackageMessagePublisher implements MessagePublisherInterface
     protected $exchange;
     protected $flags;
     protected $logger;
+    private $publisherConfirms;
+    private $timeout = 0;
+    private $lastDeliveryTag = 0;
+    private $pendingMessages = [];
 
-    public function __construct(\AMQPExchange $exchange, $flags = AMQP_NOPARAM, LoggerInterface $logger = null)
-    {
+    public function __construct(
+        \AMQPExchange $exchange,
+        $flags = AMQP_NOPARAM,
+        LoggerInterface $logger = null,
+        $publisherConfirms = false,
+        $timeout = 0
+    ) {
         $this->exchange = $exchange;
         $this->flags = $flags;
         $this->logger = $logger ?: new NullLogger();
+        $this->publisherConfirms = $publisherConfirms;
+        $this->timeout = $timeout;
+        if ($publisherConfirms) {
+            if (version_compare("1.8.0", phpversion('amqp')) === 1) {
+                throw new \Exception("Publisher confirms are not supported. Update your pecl amqp package");
+            }
+            $this->exchange->getChannel()->setConfirmCallback($this->getAckHandler(), $this->getNackHandler());
+            $this->exchange->getChannel()->confirmSelect();
+        }
     }
 
+    private function getAckHandler()
+    {
+        return function ($deliveryTag, $multiple) {
+            //remove acked from pending list
+            if ($multiple) {
+                for ($tag = 0; $tag <= $multiple; $tag ++) {
+                    unset($this->pendingMessages[$tag]);
+                }
+            } else {
+                unset($this->pendingMessages[$deliveryTag]);
+            }
+
+            if (count($this->pendingMessages) > 0) {
+                return true; //still need to wait
+            }
+            return false;
```

Comment:
Before `return false;`? My versions of php-cs-fixer (2.0-DEV) and phpcs (2.6.0) don't find anything, probably because it is an anonymous function.